### PR TITLE
More VS Code improvements

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Follow-up to <https://github.com/Homebrew/brew/pull/17934>

This PR addresses a few additional VS Code configuration things.

First, the Ruby LSP configuration now runs `brew install-bundler-gems --add-groups=style,typecheck,vscode` before starting the LSP. This should automatically install everything and keep the LSP up to date for new users.

Additionally, `editor.formatOnType` is now turned on for `ruby` files. This handles [a few things](https://github.com/Shopify/ruby-lsp/blob/main/lib/ruby_lsp/requests/on_type_formatting.rb), but a big one is that it will [automatically add `end` statements and handle indents](https://github.com/Shopify/ruby-lsp/blob/main/misc/on_type_formatting.gif). This removes the need for the `kaiwood.endwise` VS Code plugin, so I have removed it.

Finally, this PR adds the `--lsp` option to `brew typecheck` which allows us to actually enable the Sorbet plugin and view live type info (much of the code here comes from @Bo98).

### CODE REVIEW

**Review by issyl0 — APPROVED** [review]
This is fantastic!

File: Library/Homebrew/dev-cmd/typecheck.rb [hunk 1]
```diff
@@ -80,6 +85,15 @@ def run
             srb_exec << "--autocorrect"
           end
 
+          if args.lsp?
+            srb_exec << "--lsp"
+            if (watchman = which("watchman", ORIGINAL_PATHS))
+              srb_exec << "--watchman-path" << watchman
+            else
+              srb_exec << "--disable-watchman" unless which("watchman", ORIGINAL_PATHS)
```
Comment (Bo98) [inline Library/Homebrew/dev-cmd/typecheck.rb:93]:
```suggestion
              srb_exec << "--disable-watchman"
```

already in an `else` block

**Review by Bo98 — APPROVED** [review]

File: Library/Homebrew/dev-cmd/typecheck.rb (hunk 1 above)
Comment (Rylan12) [reply Library/Homebrew/dev-cmd/typecheck.rb:93]: Oops, good catch!

**Comment by MikeMcQuaid** [comment]
Great work, thanks @Rylan12!

**Pull request merged**